src/Swashbuckle.AspNetCore.Cli/Program.cs

```diff
@@ -156,9 +126,103 @@ public static int Main(string[] args)
                 });
             });
 
+            // > dotnet swagger list
+            runner.SubCommand("list", "retrieves the list of Swagger document names from a startup assembly", c =>
+            {
+                c.Argument("startupassembly", "relative path to the application's startup assembly");
+                c.Option("--output", "relative path where the document names will be output, defaults to stdout");
+                c.OnRun((namedArgs) =>
+                {
+                    string subProcessCommandLine = PrepareCommandLine(args, namedArgs);
+
+                    var subProcess = Process.Start("dotnet", subProcessCommandLine);
+
+                    subProcess.WaitForExit();
+                    return subProcess.ExitCode;
+                });
+            });
+
+            // > dotnet swagger _list ... (* should only be invoked via "dotnet exec")
+            runner.SubCommand("_list", "", c =>
+            {
+                c.Argument("startupassembly", "");
+                c.Option("--output", "");
+                c.OnRun((namedArgs) =>
+                {
+                    SetupAndRetrieveSwaggerProviderAndOptions(namedArgs, out var swaggerProvider, out var swaggerOptions);
+                    var docMetaProvider = swaggerProvider as ISwaggerDocumentMetadataProvider;
+                    var docNames = docMetaProvider.GetDocumentNames();
```

Comment:
If `docMetaProvider` is null for some reason, we should log something to the console and bail out gracefully.